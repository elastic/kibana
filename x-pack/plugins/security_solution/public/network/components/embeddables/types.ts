/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import type { RenderTooltipContentParams } from '../../../../../maps/public';

export interface IndexPatternMapping {
  title: string;
  id: string;
}

export interface LayerMappingDetails {
  metricField: string;
  geoField: string;
  tooltipProperties: string[];
  label: string;
}

export interface LayerMapping {
  source: LayerMappingDetails;
  destination: LayerMappingDetails;
}

export interface LayerMappingCollection {
  [indexPatternTitle: string]: LayerMapping;
}

export interface MapFeature {
  id: number;
  layerId: string;
}

export interface FeatureGeometry {
  coordinates: [number];
  type: string;
}

export type MapToolTipProps = Partial<RenderTooltipContentParams>;
