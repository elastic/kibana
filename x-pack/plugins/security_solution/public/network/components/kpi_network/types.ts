/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { UpdateDateRange } from '../../../common/components/charts/common';
import { GlobalTimeArgs } from '../../../common/containers/use_global_time';

export interface NetworkKpiProps {
  filterQuery: string;
  from: string;
  indexNames: string[];
  to: string;
  narrowDateRange: UpdateDateRange;
  setQuery: GlobalTimeArgs['setQuery'];
  skip: boolean;
}
