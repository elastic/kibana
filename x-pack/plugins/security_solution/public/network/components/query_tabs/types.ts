/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import type { DataViewBase } from '@kbn/es-query';
import { DocValueFields } from '../../../common/containers/source';
import { GlobalTimeArgs } from '../../../common/containers/use_global_time';
import { networkModel } from '../../store';
import { ESTermQuery } from '../../../../common/typed_json';
import { NarrowDateRange } from '../../../common/components/ml/types';
import { FlowTarget } from '../../../../common/search_strategy';

interface QueryTabBodyProps extends Pick<GlobalTimeArgs, 'setQuery' | 'deleteQuery'> {
  endDate: string;
  filterQuery?: string | ESTermQuery;
  indexNames: string[];
  ip?: string;
  narrowDateRange?: NarrowDateRange;
  skip: boolean;
  startDate: string;
  type: networkModel.NetworkType;
}

export type NetworkComponentQueryProps = QueryTabBodyProps & {
  docValueFields?: DocValueFields[];
};

export type IPsQueryTabBodyProps = QueryTabBodyProps & {
  flowTarget: FlowTarget;
  indexPattern: DataViewBase;
};

export type FTQueryTabBodyProps = QueryTabBodyProps & {
  flowTarget: FlowTarget;
};

export type IPQueryTabBodyProps = FTQueryTabBodyProps & {
  ip: string;
};

export type HttpQueryTabBodyProps = QueryTabBodyProps;
