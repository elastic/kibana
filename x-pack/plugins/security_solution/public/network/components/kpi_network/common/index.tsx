/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import React from 'react';

import { EuiFlexItem, EuiLoadingSpinner, EuiFlexGroup } from '@elastic/eui';
import styled from 'styled-components';

import { manageQuery } from '../../../../common/components/page/manage_query';
import { NetworkKpiStrategyResponse } from '../../../../../common/search_strategy';
import {
  StatItemsComponent,
  StatItemsProps,
  useKpiMatrixStatus,
  StatItems,
} from '../../../../common/components/stat_items';
import { UpdateDateRange } from '../../../../common/components/charts/common';

const kpiWidgetHeight = 228;

export const FlexGroup = styled(EuiFlexGroup)`
  min-height: ${kpiWidgetHeight}px;
`;

FlexGroup.displayName = 'FlexGroup';

export const NetworkKpiBaseComponent = React.memo<{
  fieldsMapping: Readonly<StatItems[]>;
  data: NetworkKpiStrategyResponse;
  loading?: boolean;
  id: string;
  from: string;
  to: string;
  narrowDateRange: UpdateDateRange;
}>(({ fieldsMapping, data, id, loading = false, from, to, narrowDateRange }) => {
  const statItemsProps: StatItemsProps[] = useKpiMatrixStatus(
    fieldsMapping,
    data,
    id,
    from,
    to,
    narrowDateRange
  );

  if (loading) {
    return (
      <FlexGroup justifyContent="center" alignItems="center">
        <EuiFlexItem grow={false}>
          <EuiLoadingSpinner size="xl" />
        </EuiFlexItem>
      </FlexGroup>
    );
  }

  return (
    <EuiFlexGroup wrap>
      {statItemsProps.map((mappedStatItemProps) => (
        <StatItemsComponent {...mappedStatItemProps} />
      ))}
    </EuiFlexGroup>
  );
});

NetworkKpiBaseComponent.displayName = 'NetworkKpiBaseComponent';

export const NetworkKpiBaseComponentManage = manageQuery(NetworkKpiBaseComponent);
