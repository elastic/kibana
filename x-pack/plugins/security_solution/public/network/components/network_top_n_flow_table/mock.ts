/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import {
  NetworkTopNFlowStrategyResponse,
  FlowTargetSourceDest,
} from '../../../../common/search_strategy';

export const mockData: NetworkTopNFlowStrategyResponse = {
  totalCount: 524,
  edges: [
    {
      node: {
        source: {
          autonomous_system: {
            name: 'Google, Inc',
            number: 15169,
          },
          domain: ['test.domain.com'],
          flows: 12345,
          destination_ips: 12,
          ip: '8.8.8.8',
          location: {
            geo: {
              continent_name: ['North America'],
              country_iso_code: ['US'],
              city_name: ['Mountain View'],
              region_iso_code: ['US-CA'],
              region_name: ['California'],
            },
            flowTarget: FlowTargetSourceDest.source,
          },
        },
        network: {
          bytes_in: 3826633497,
          bytes_out: 1083495734,
        },
      },
      cursor: {
        value: '8.8.8.8',
      },
    },
    {
      node: {
        source: {
          autonomous_system: {
            name: 'TM Net, Internet Service Provider',
            number: 4788,
          },
          domain: ['test.domain.net', 'test.old.domain.net'],
          flows: 12345,
          destination_ips: 12,
          ip: '9.9.9.9',
          location: {
            geo: {
              continent_name: ['Asia'],
              country_iso_code: ['MY'],
              city_name: ['Petaling Jaya'],
              region_iso_code: ['MY-10'],
              region_name: ['Selangor'],
            },
            flowTarget: FlowTargetSourceDest.source,
          },
        },
        network: {
          bytes_in: 3826633497,
          bytes_out: 1083495734,
        },
      },
      cursor: {
        value: '9.9.9.9',
      },
    },
  ],
  pageInfo: {
    activePage: 1,
    fakeTotalCount: 50,
    showMorePagesIndicator: true,
  },
  rawResponse: {} as NetworkTopNFlowStrategyResponse['rawResponse'],
};
