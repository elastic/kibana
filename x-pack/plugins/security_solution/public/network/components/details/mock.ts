/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import { NetworkDetailsStrategyResponse } from '../../../../common/search_strategy';

export const mockData: Readonly<
  Record<string, NetworkDetailsStrategyResponse['networkDetails']>
> = {
  complete: {
    source: {
      firstSeen: '2019-02-07T17:19:41.636Z',
      lastSeen: '2019-02-07T17:19:41.636Z',
      autonomousSystem: { organization: { name: 'Test Org' }, number: 12345 },
      geo: {
        continent_name: ['North America'],
        city_name: ['New York'],
        country_iso_code: ['US'],
        country_name: undefined,
        location: {
          lat: [40.7214],
          lon: [-74.0052],
        },
        region_iso_code: ['US-NY'],
        region_name: ['New York'],
      },
    },
    destination: {
      firstSeen: '2019-02-07T17:19:41.648Z',
      lastSeen: '2019-02-07T17:19:41.648Z',
      autonomousSystem: { organization: { name: 'Test Org' }, number: 12345 },
      geo: {
        continent_name: ['North America'],
        city_name: ['New York'],
        country_iso_code: ['US'],
        country_name: undefined,
        location: {
          lat: [40.7214],
          lon: [-74.0052],
        },
        region_iso_code: ['US-NY'],
        region_name: ['New York'],
      },
    },
    host: {
      os: {
        kernel: ['4.14.50-v7+'],
        name: ['Raspbian GNU/Linux'],
        family: [''],
        version: ['9 (stretch)'],
        platform: ['raspbian'],
      },
      name: ['raspberrypi'],
      id: ['b19a781f683541a7a25ee345133aa399'],
      ip: ['10.10.10.10'],
      architecture: ['armv7l'],
    },
  },
};
