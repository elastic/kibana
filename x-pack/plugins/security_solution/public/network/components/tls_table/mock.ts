/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License;
 * you may not use this file except in compliance with the Elastic License.
 */

import { NetworkTlsStrategyResponse } from '../../../../common/search_strategy';

export const mockTlsData: NetworkTlsStrategyResponse = {
  totalCount: 2,
  edges: [
    {
      node: {
        _id: '2fe3bdf168af35b9e0ce5dc583bab007c40d47de',
        subjects: ['*.elastic.co'],
        ja3: ['7851693188210d3b271aa1713d8c68c2', 'fb4726d465c5f28b84cd6d14cedd13a7'],
        issuers: ['DigiCert SHA2 Secure Server CA'],
        notAfter: ['2021-04-22T12:00:00.000Z'],
      },
      cursor: {
        value: '2fe3bdf168af35b9e0ce5dc583bab007c40d47de',
      },
    },
    {
      node: {
        _id: '61749734b3246f1584029deb4f5276c64da00ada',
        subjects: ['api.snapcraft.io'],
        ja3: ['839868ad711dc55bde0d37a87f14740d'],
        issuers: ['DigiCert SHA2 Secure Server CA'],
        notAfter: ['2019-05-22T12:00:00.000Z'],
      },
      cursor: {
        value: '61749734b3246f1584029deb4f5276c64da00ada',
      },
    },
    {
      node: {
        _id: '6560d3b7dd001c989b85962fa64beb778cdae47a',
        subjects: ['changelogs.ubuntu.com'],
        ja3: ['da12c94da8021bbaf502907ad086e7bc'],
        issuers: ["Let's Encrypt Authority X3"],
        notAfter: ['2019-06-27T01:09:59.000Z'],
      },
      cursor: {
        value: '6560d3b7dd001c989b85962fa64beb778cdae47a',
      },
    },
  ],
  pageInfo: {
    activePage: 1,
    fakeTotalCount: 50,
    showMorePagesIndicator: true,
  },
  rawResponse: {} as NetworkTlsStrategyResponse['rawResponse'],
};
