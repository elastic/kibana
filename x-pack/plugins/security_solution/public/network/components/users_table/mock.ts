/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import { NetworkUsersStrategyResponse } from '../../../../common/search_strategy';

export const mockUsersData: NetworkUsersStrategyResponse = {
  edges: [
    {
      node: {
        _id: '_apt',
        user: {
          id: ['104'],
          name: '_apt',
          groupId: ['65534'],
          groupName: ['nogroup'],
          count: 10,
        },
      },
      cursor: {
        value: '_apt',
        tiebreaker: null,
      },
    },
    {
      node: {
        _id: 'root',
        user: {
          id: ['0'],
          name: 'root',
          groupId: ['116', '0'],
          groupName: ['Debian-exim', 'root'],
          count: 108,
        },
      },
      cursor: {
        value: 'root',
        tiebreaker: null,
      },
    },
    {
      node: {
        _id: 'systemd-resolve',
        user: {
          id: ['102'],
          name: 'systemd-resolve',
          groupId: [],
          groupName: [],
          count: 4,
        },
      },
      cursor: {
        value: 'systemd-resolve',
        tiebreaker: null,
      },
    },
  ],
  totalCount: 3,
  pageInfo: {
    activePage: 1,
    fakeTotalCount: 3,
    showMorePagesIndicator: true,
  },
  rawResponse: {} as NetworkUsersStrategyResponse['rawResponse'],
};
