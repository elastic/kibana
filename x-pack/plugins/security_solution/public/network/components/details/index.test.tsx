/*
 * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
 * or more contributor license agreements. Licensed under the Elastic License
 * 2.0; you may not use this file except in compliance with the Elastic License
 * 2.0.
 */

import { shallow } from 'enzyme';
import React from 'react';
import { ActionCreator } from 'typescript-fsa';

import { FlowTarget } from '../../../graphql/types';
import '../../../common/mock/match_media';
import {
  apolloClientObservable,
  mockGlobalState,
  TestProviders,
  SUB_PLUGINS_REDUCER,
  kibanaObservable,
  createSecuritySolutionStorageMock,
} from '../../../common/mock';
import { createStore, State } from '../../../common/store';
import { networkModel } from '../../store';

import { IpOverview } from './index';
import { mockData } from './mock';
import { mockAnomalies } from '../../../common/components/ml/mock';
import { NarrowDateRange } from '../../../common/components/ml/types';

describe('IP Overview Component', () => {
  const state: State = mockGlobalState;

  const { storage } = createSecuritySolutionStorageMock();
  let store = createStore(
    state,
    SUB_PLUGINS_REDUCER,
    apolloClientObservable,
    kibanaObservable,
    storage
  );

  beforeEach(() => {
    store = createStore(
      state,
      SUB_PLUGINS_REDUCER,
      apolloClientObservable,
      kibanaObservable,
      storage
    );
  });

  describe('rendering', () => {
    const mockProps = {
      anomaliesData: mockAnomalies,
      data: mockData.IpOverview,
      endDate: '2019-06-18T06:00:00.000Z',
      flowTarget: FlowTarget.source,
      loading: false,
      id: 'ipOverview',
      ip: '10.10.10.10',
      isInDetailsSidePanel: false,
      isLoadingAnomaliesData: false,
      narrowDateRange: (jest.fn() as unknown) as NarrowDateRange,
      startDate: '2019-06-15T06:00:00.000Z',
      type: networkModel.NetworkType.details,
      updateFlowTargetAction: (jest.fn() as unknown) as ActionCreator<{
        flowTarget: FlowTarget;
      }>,
    };

    test('it renders the default IP Overview', () => {
      const wrapper = shallow(
        <TestProviders store={store}>
          <IpOverview {...mockProps} />
        </TestProviders>
      );

      expect(wrapper.find('IpOverview')).toMatchSnapshot();
    });

    test('it renders the side panel IP overview', () => {
      const panelViewProps = {
        ...mockProps,
        isInDetailsSidePanel: true,
      };
      const wrapper = shallow(
        <TestProviders store={store}>
          <IpOverview {...panelViewProps} />
        </TestProviders>
      );

      expect(wrapper.find('IpOverview')).toMatchSnapshot();
    });
  });
});
