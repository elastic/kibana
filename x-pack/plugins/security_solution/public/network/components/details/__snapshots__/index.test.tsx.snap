// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IP Overview Component rendering it renders the default IP Overview 1`] = `
<IpOverview
  anomaliesData={
    Object {
      "anomalies": Array [
        Object {
          "detectorIndex": 0,
          "entityName": "process.name",
          "entityValue": "du",
          "influencers": Array [
            Object {
              "host.name": "zeek-iowa",
            },
            Object {
              "process.name": "du",
            },
            Object {
              "user.name": "root",
            },
          ],
          "jobId": "job-1",
          "rowId": "1561157194802_0",
          "severity": 16.193669439507826,
          "source": Object {
            "actual": Array [
              1,
            ],
            "bucket_span": 900,
            "by_field_name": "process.name",
            "by_field_value": "du",
            "detector_index": 0,
            "function": "rare",
            "function_description": "rare",
            "influencers": Array [
              Object {
                "influencer_field_name": "user.name",
                "influencer_field_values": Array [
                  "root",
                ],
              },
              Object {
                "influencer_field_name": "process.name",
                "influencer_field_values": Array [
                  "du",
                ],
              },
              Object {
                "influencer_field_name": "host.name",
                "influencer_field_values": Array [
                  "zeek-iowa",
                ],
              },
            ],
            "initial_record_score": 16.193669439507826,
            "is_interim": false,
            "job_id": "job-1",
            "multi_bucket_impact": 0,
            "partition_field_name": "host.name",
            "partition_field_value": "zeek-iowa",
            "probability": 0.024041164411288146,
            "record_score": 16.193669439507826,
            "result_type": "record",
            "timestamp": 1560664800000,
            "typical": Array [
              0.024041164411288146,
            ],
          },
          "time": 1560664800000,
        },
        Object {
          "detectorIndex": 0,
          "entityName": "process.name",
          "entityValue": "ls",
          "influencers": Array [
            Object {
              "host.name": "zeek-iowa",
            },
            Object {
              "process.name": "ls",
            },
            Object {
              "user.name": "root",
            },
          ],
          "jobId": "job-2",
          "rowId": "1561157194802_1",
          "severity": 16.193669439507826,
          "source": Object {
            "actual": Array [
              1,
            ],
            "bucket_span": 900,
            "by_field_name": "process.name",
            "by_field_value": "ls",
            "detector_index": 0,
            "function": "rare",
            "function_description": "rare",
            "influencers": Array [
              Object {
                "influencer_field_name": "user.name",
                "influencer_field_values": Array [
                  "root",
                ],
              },
              Object {
                "influencer_field_name": "process.name",
                "influencer_field_values": Array [
                  "ls",
                ],
              },
              Object {
                "influencer_field_name": "host.name",
                "influencer_field_values": Array [
                  "zeek-iowa",
                ],
              },
            ],
            "initial_record_score": 16.193669439507826,
            "is_interim": false,
            "job_id": "job-2",
            "multi_bucket_impact": 0,
            "partition_field_name": "host.name",
            "partition_field_value": "zeek-iowa",
            "probability": 0.024041164411288146,
            "record_score": 16.193669439507826,
            "result_type": "record",
            "timestamp": 1560664800000,
            "typical": Array [
              0.024041164411288146,
            ],
          },
          "time": 1560664800000,
        },
      ],
      "interval": "day",
    }
  }
  endDate="2019-06-18T06:00:00.000Z"
  flowTarget="source"
  id="ipOverview"
  ip="10.10.10.10"
  isInDetailsSidePanel={false}
  isLoadingAnomaliesData={false}
  loading={false}
  narrowDateRange={[MockFunction]}
  startDate="2019-06-15T06:00:00.000Z"
  type="details"
  updateFlowTargetAction={[MockFunction]}
/>
`;

exports[`IP Overview Component rendering it renders the side panel IP overview 1`] = `
<IpOverview
  anomaliesData={
    Object {
      "anomalies": Array [
        Object {
          "detectorIndex": 0,
          "entityName": "process.name",
          "entityValue": "du",
          "influencers": Array [
            Object {
              "host.name": "zeek-iowa",
            },
            Object {
              "process.name": "du",
            },
            Object {
              "user.name": "root",
            },
          ],
          "jobId": "job-1",
          "rowId": "1561157194802_0",
          "severity": 16.193669439507826,
          "source": Object {
            "actual": Array [
              1,
            ],
            "bucket_span": 900,
            "by_field_name": "process.name",
            "by_field_value": "du",
            "detector_index": 0,
            "function": "rare",
            "function_description": "rare",
            "influencers": Array [
              Object {
                "influencer_field_name": "user.name",
                "influencer_field_values": Array [
                  "root",
                ],
              },
              Object {
                "influencer_field_name": "process.name",
                "influencer_field_values": Array [
                  "du",
                ],
              },
              Object {
                "influencer_field_name": "host.name",
                "influencer_field_values": Array [
                  "zeek-iowa",
                ],
              },
            ],
            "initial_record_score": 16.193669439507826,
            "is_interim": false,
            "job_id": "job-1",
            "multi_bucket_impact": 0,
            "partition_field_name": "host.name",
            "partition_field_value": "zeek-iowa",
            "probability": 0.024041164411288146,
            "record_score": 16.193669439507826,
            "result_type": "record",
            "timestamp": 1560664800000,
            "typical": Array [
              0.024041164411288146,
            ],
          },
          "time": 1560664800000,
        },
        Object {
          "detectorIndex": 0,
          "entityName": "process.name",
          "entityValue": "ls",
          "influencers": Array [
            Object {
              "host.name": "zeek-iowa",
            },
            Object {
              "process.name": "ls",
            },
            Object {
              "user.name": "root",
            },
          ],
          "jobId": "job-2",
          "rowId": "1561157194802_1",
          "severity": 16.193669439507826,
          "source": Object {
            "actual": Array [
              1,
            ],
            "bucket_span": 900,
            "by_field_name": "process.name",
            "by_field_value": "ls",
            "detector_index": 0,
            "function": "rare",
            "function_description": "rare",
            "influencers": Array [
              Object {
                "influencer_field_name": "user.name",
                "influencer_field_values": Array [
                  "root",
                ],
              },
              Object {
                "influencer_field_name": "process.name",
                "influencer_field_values": Array [
                  "ls",
                ],
              },
              Object {
                "influencer_field_name": "host.name",
                "influencer_field_values": Array [
                  "zeek-iowa",
                ],
              },
            ],
            "initial_record_score": 16.193669439507826,
            "is_interim": false,
            "job_id": "job-2",
            "multi_bucket_impact": 0,
            "partition_field_name": "host.name",
            "partition_field_value": "zeek-iowa",
            "probability": 0.024041164411288146,
            "record_score": 16.193669439507826,
            "result_type": "record",
            "timestamp": 1560664800000,
            "typical": Array [
              0.024041164411288146,
            ],
          },
          "time": 1560664800000,
        },
      ],
      "interval": "day",
    }
  }
  endDate="2019-06-18T06:00:00.000Z"
  flowTarget="source"
  id="ipOverview"
  ip="10.10.10.10"
  isInDetailsSidePanel={true}
  isLoadingAnomaliesData={false}
  loading={false}
  narrowDateRange={[MockFunction]}
  startDate="2019-06-15T06:00:00.000Z"
  type="details"
  updateFlowTargetAction={[MockFunction]}
/>
`;
